src/debugger/handler_dbgp.c

```diff
@@ -319,7 +319,7 @@ static void send_message_ex(xdebug_con *context, xdebug_xml_node *message, int s
 	tmp = make_message(context, message);
 	if ((size_t) SSENDL(context->socket, tmp->d, tmp->l) != tmp->l) {
 		char *sock_error = php_socket_strerror(php_socket_errno(), NULL, 0);
-		char *utime_str = xdebug_sprintf("%F", xdebug_get_utime());
+		char *utime_str = xdebug_sprintf("%F", xdebug_get_nanotime() / (double)NANOS_IN_SEC);
```

Comment:
I see this division with `/ (double)NANOS_IN_SEC` done a lot, is there are cleaner way to do this? Especially the `(double)` cast irks me.

In a whole bunch of other places there is `(x - XG_BASE(start_nano_time) / (double)NANOS_IN_SEC`, which I think could be ripe for a macro:
```
#define XDEBUG_SECONDS_SINCE_START(t) (((t) - XG_BASE(start_nano_time) / (double)NANOS_IN_SEC)
```

Reply:
solved